coalib/settings/Section.py

```diff
@@ -43,6 +46,53 @@ def append_to_sections(sections,
         key, str(value), origin, from_cli=from_cli, to_append=to_append))
 
 
+def extract_aspects_from_section(section):
+    """
+    Extract aspects and its related setting from a section and create an
+    aspectlist from it.
+
+    :section: Section object.
+    :return:  aspectlist containing aspectclass instance with user-defined
+              tastes.
+    """
+    if not len(section.get('aspects')):
+        # Off switch for aspect-based configuration
+        return None
+    if not len(section.get('language')):
+        logging.error('Language was not found in configuration file. '
+                      'Usage of aspect-based configuration must include'
+                      'language information.')
+        raise AttributeError
+        # TODO acquire language from interactive input instead of just raising
+        # error.
+
+    aspects = aspectlist(section.get('aspects'))
+    language = section.get('language')
+    try:
+        Language[language]
+    except AttributeError:
+        logging.error('Language {} is not a valid language name or is not '
+                      'recognized by coala.'.format(language))
+        raise AttributeError
+
+    aspects_instance = aspectlist()
+    for aspect in aspects:
+        tastes_to_init = dict()
+        for taste in aspect.tastes:
+            # TODO handle duplicate taste meant for different aspect
+            try:
+                tastes_to_init[taste] = section[taste]
+            except IndexError:
+                pass
+        aspects_instance.append(aspect(language, **tastes_to_init))
+
+    # TODO exclude aspects
+
+    # We don't need it again, so remove it to avoid name conflict.
+    section.delete_setting('aspects')
```

Comment:
This should not be in an "extract .." method.

The caller should delete the section.